include/openPMD/Iteration.hpp

```diff
@@ -97,7 +111,21 @@ class Iteration : public Attributable
     void flushFileBased(std::string const&, uint64_t);
     void flushGroupBased(uint64_t);
     void flush();
-    void read();
+    void
+    read();
+
+    /*
+     * Indicates that an iteration has been logically closed.
+     * Will be physically closed upon next flush.
+     */
+    std::shared_ptr< bool > m_closed = std::make_shared< bool >( false );
```

Comment:
Can you quickly remind why this cannot be a member? Is some lifetime of the series/iteration preventing this?

Reply:
`class Iteration` is designed as a handle, all relevant information should be shared between copies.
```cpp
auto iteration0 = series.iterations[0];
iteration0.close();
```
If the flag were not behind a `shared_ptr`, we would then get:
```cpp
iteration0.m_closed == true
// but
series.iterations[0].m_closed == false
```
(This touches a broader point that we should keep in mind, if we renovate the frontend some day: While most classes in the openPMD API are handles, this is implemented rather in an ad-hoc way currently. I would prefer a design similar to ADIOS2, i.e. have some none-copyable resource like `internal::Iteration` and a user-facing handle `external::Iteration` that acts like a `shared_ptr` to `internal::Iteration`)